pkg/apis/v1alpha5/suite_test.go

```diff
@@ -521,3 +524,55 @@ var _ = Describe("Limits", func() {
 		Expect(provisioner.Spec.Limits.ExceededBy(provisioner.Status.Resources)).To(MatchError("cpu resource usage of 17 exceeds limit of 16"))
 	})
 })
+
+var _ = Describe("Provisioner Annotation", func() {
+	var provisioner *Provisioner
+	BeforeEach(func() {
+		provisioner = &Provisioner{
+			ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+			Spec: ProvisionerSpec{
+				Taints: []v1.Taint{
+					{
+						Key:    "key",
+						Effect: v1.TaintEffectNoExecute,
+					},
+				},
+				StartupTaints: []v1.Taint{
+					{
+						Key:    "key",
+						Effect: v1.TaintEffectNoExecute,
+					},
+				},
+				Labels: map[string]string{
+					"keyLabel": "valueLabel",
+				},
+				KubeletConfiguration: &KubeletConfiguration{
+					MaxPods: ptr.Int32(10),
+				},
+				Annotations: map[string]string{
+					"keyAnnotation": "valueAnnotation",
+				},
+			},
+		}
+	})
+	It("should only change hash when static felids are updated", func() {
+		actualHash, err := provisioner.StaticDriftHash()
+		Expect(err).ToNot(HaveOccurred())
+
+		expectedHash, err := hashstructure.Hash(provisioner, hashstructure.FormatV2, &hashstructure.HashOptions{
+			SlicesAsSets:    true,
+			IgnoreZeroValue: true,
+			ZeroNil:         true,
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(actualHash).To(Equal(fmt.Sprint(expectedHash)))
+
+		// Update a behavior field
+		provisioner.Spec.Consolidation = &Consolidation{Enabled: lo.ToPtr(true)}
```

Comment:
Can you change ALL the ignored fields here, just to be sure? 

Can you also just do a sanity check that the way a map is defined won't affect this? Maps aren't ordered, but I want to make sure that the hashing library does this correctly.